tests/integrationv2/test_client_authentication.py

```diff
@@ -118,3 +118,111 @@ def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, ci
         assert results.exception is None
         assert results.exit_code == 255
         assert b'Error: Mutual Auth was required, but not negotiated' in results.stderr
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", [cipher for cipher in ALL_TEST_CIPHERS if 'ECDSA' not in cipher.name], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_CURVES)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_CERTS, ids=get_parameter_name)
+def test_client_auth_with_s2n_client_no_cert(managed_process, cipher, provider, curve, protocol, certificate):
```

Comment:
This test share a lot of common code with test_client_auth_with_s2n_client_with_cert. Can we factor out any of the duplication?

Reply:
i could rework some of it but i was trying to follow the precedent in test_happy_path